### PR TITLE
Fixed animation part

### PR DESCRIPTION
**Pull request by Jacbo1**
- Rate setting now multiplies playback rate so a Rate of 1 is now actually 1x playback speed
- Fixed divide by 0 when Min and Max settings are the same number (doing this now explicitly sets the progress of the animation)
- Setting Rate to 0 now pauses animation like one would expect instead of also setting the current frame to 0
- Removed rate limit that was sometimes so low that animations could not be played at the original speed
- Made self.frame wrap from 0 to 1 so extremely high rates will never become so high they experience floating point precision loss
- Removed some unnecessary code that never ran
- Replaced some code with more optimized alternatives

Everything has been tested by me

### CODE REVIEW

**Comment by Jacbo1** [comment]
I just found out that pull requests are usually made to the develop branch so I'll redo this for that branch.

**Comment by thegrb93** [comment]
Ah yeah. Just do `git rebase develop`

**Comment by thegrb93** [comment]
You can use the edit button in the top right to change the target branch next time instead of closing it.

File: lua/pac3/core/client/parts/animation.lua [hunk 1]
```diff
@@ -244,59 +244,55 @@ end
...
-	if seq >= 0 and seq < count and count > 0 then
-		duration = ent:SequenceDuration(seq)
-	else
+	if seq < 0 or seq >= count then
 		-- It's an invalid sequence. Don't bother
 		return
 	end
+	
+	local min = self.Min
+	local max = self.Max
```
Comment (thegrb93) [inline lua/pac3/core/client/parts/animation.lua:256]: You could also do `local length = max - min` to optimize away the `max - min` in the later code

**Comment by Jacbo1** [comment]
Well this doesn't directly port to the develop branch

File: lua/pac3/core/client/parts/animation.lua [hunk 2]
```diff
@@ -244,59 +244,55 @@ end
...
+		local cycle = min + math.abs(math.Round(self.frame + self.Offset * 2) - self.frame - self.Offset * 2) * 2 * (max - min)
 
 		if pac.IsNumberValid(cycle) then
 			ent:SetCycle(not self.InvertFrames and cycle or (1 - cycle))
 		end
 	else
-		self.frame = self.frame + rate
-		local cycle = min + ((self.frame + self.Offset) * 0.5) % 1 * (max - min)
+		self.frame = (self.frame + rate) % 1
+		local cycle = min + (self.frame + self.Offset * 2) % 1 * (max - min)
```
Comment (thegrb93) [inline lua/pac3/core/client/parts/animation.lua:290]: What's the *2 do here?

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/pac3/core/client/parts/animation.lua (hunk 2 above)
Comment (Jacbo1) [reply lua/pac3/core/client/parts/animation.lua:290]: I kinda trialed and errored until everything worked right. It's all been tested (on the master branch version)

Comment (Jacbo1) [reply lua/pac3/core/client/parts/animation.lua:290]: It could actually be a mistake from when I was testing with min = 0.5

**Comment by Jacbo1** [comment]
I finished the develop branch version and also fixed something I didn't catch before so I'll close this and make a new one since this is also pulling from a different branch I made.

**Pull request closed without merging**